modules/data.land/R/cohort2pool.R

```diff
@@ -19,33 +19,55 @@
 
 cohort2pool <- function(dat, allom_param = NULL, dbh_name="DBH") {
   
-  ## Grab DBH
-  dbh <- dat[[2]][,dbh_name]
   #Grab plot size
-  plot_size <- dat[[1]]$area
+  herb_plot <- dat[[1]]$clipArea[1]
   #Grab number of plots
-  plot_num <- length(unique(paste(dat[[2]]$site_name,dat[[2]]$plot,dat[[2]]$Subplot)))
-  
-  ## Grab allometry
-  if(is.null(allom_param)){
-    a <- -2.0127                        
-    b <- 2.4342
-    biomass = exp(a + b*log(dbh))
-    #Hard code foliage equation from Jenkins paper
-    b0 <- -4.0813
-    b1 <- 5.8816
-    ratio = ifelse(dbh>=2.5,exp(b0 + (b1/dbh)),exp(b0 + (b1/2.5)))
-  } else {
-    #Predict AGB using allom.predit code taken from Allom.Vignette.Rmd
-    # allom.fit = #outputs from AllomAve function
-    # stand = allom.predict(allom.fit,dbh = dbh,pft = "LH",component = 3,use = "Bg",interval = "prediction")
-    # AGB = apply(stand,1,sum)
-    # hist(AGB)
-    AGB <- NULL
-    print("user provided allometry parameters not yet supported")
-    #return(NULL)
-    return(AGB)
+  herb_num <- length(unique(dat[[1]]$plot))
+  #
+  if(sum(!is.na(dat[[2]]))==0){
+    biomass <- 0
+    total_area <- 1
+    ratio <- 0
+  }else{
+    ## Grab DBH
+    dbh <- dat[[2]][,dbh_name]
+    
+    #calculate total area
+    unique_subplot <- unique(paste(dat[[2]]$site_name,dat[[2]]$plot,dat[[2]]$Subplot))
+    area <- c()
+    for (i in 1:length(unique_subplot)) {
+      subplot_ID <- strsplit(unique_subplot[i], " ")[[1]][3]
+      if(subplot_ID %in% c(31, 32, 40, 41)){
+        area <- c(area, 100)
+      }else if(subplot_ID %in% c(21, 23, 39)){
```

Comment:
This mapping between subplot_ID and plot area seems suspicious to me -- is this relationship between ID and area really universal across all NEON sites? Could you point me to the documentation that confirms this?